upf/upf_cli.c

```diff
@@ -635,6 +641,85 @@ VLIB_CLI_COMMAND (upf_show_spec_release_command, static) =
 };
 /* *INDENT-ON* */
 
+static clib_error_t *
+upf_node_id_command_fn (vlib_main_t * vm,
+			unformat_input_t * main_input,
+			vlib_cli_command_t * cmd)
+{
+  upf_main_t *gtm = &upf_main;
+  unformat_input_t _line_input, *line_input = &_line_input;
+  clib_error_t *error = NULL;
+  ip46_address_t ip46;
+  u8 *fqdn = 0;
+  pfcp_node_id_t *node_id = &gtm->node_id;
+
+  if (node_id->type == NID_FQDN)
+    vec_free (node_id->fqdn);
```

Comment:
this can't go here. The function might abort without setting a new value. The old value needs to remain valid in that case.

Reply:
Fixed this by using pfcp free_node_id func only if valid data is met in CLI command